ros/src/waypoint_follower/src/pure_pursuit_core.cpp

```diff
@@ -251,23 +251,9 @@ bool PurePursuit::verifyFollowing() const
 }
 geometry_msgs::Twist PurePursuit::calcTwist(double curvature, double cmd_velocity) const
 {
-  // verify whether vehicle is following the path
-  bool following_flag = verifyFollowing();
-  static double prev_angular_velocity = 0;
-
   geometry_msgs::Twist twist;
   twist.linear.x = cmd_velocity;
-  if (!following_flag)
-  {
-    //ROS_ERROR_STREAM("Not following");
-    twist.angular.z = current_velocity_.twist.linear.x * curvature;
-  }
-  else
-  {
-    twist.angular.z = prev_angular_velocity;
-  }
-
-  prev_angular_velocity = twist.angular.z;
+  twist.angular.z = current_velocity_.twist.linear.x * curvature;
```

Comment:
following_flag - indicates whether car is following the waypoints and if it does no changes are done in the control. removing it means that car is following the waypoints without weaving